connect/json/src/main/java/org/apache/kafka/connect/json/JsonConverter.java

```diff
@@ -285,7 +288,7 @@ public JsonConverter() {
         // floating point numbers that cannot fit into float64
         final Set<DeserializationFeature> deserializationFeatures = new HashSet<>();
         deserializationFeatures.add(DeserializationFeature.USE_BIG_DECIMAL_FOR_FLOATS);
-        deserializer = new JsonDeserializer(deserializationFeatures);
+        deserializer = new JsonDeserializer(deserializationFeatures, true);
```

Comment:
Passing in a `JsonNodeFactory` instance might be better here, because then the `JsonConverter` instances and `JsonDeserializer` in each converter will all use the same `JsonNodeFactory`.

As it currently stands, the `JsonDeserializer` has it's own instance of the `JsonNodeFactory`, and it's possible that they could be set up differently in the future and not caught.

Reply:
On it...

Reply:
What do you think about also enabling `SerializationFeature.WRITE_BIGDECIMAL_AS_PLAIN`?

This would mean serialiser won't write numbers using scientific notation, e.g. `100` , to `1E+2`.  Both are valid JSON, and the later takes up less space, but can confuse some users and maybe some downstream systems.

Reply:
Discussion offline - not going with none SN as this woudl require a KIP